concrete/src/File/EditResponse.php

```diff
@@ -1,22 +1,55 @@
 <?php
+
 namespace Concrete\Core\File;
 
+use Concrete\Core\Application\EditResponse as CoreEditResponse;
 use Concrete\Core\Entity\File\File as FileEntity;
 
-class EditResponse extends \Concrete\Core\Application\EditResponse
+/**
+ * The result of file import.
+ */
+class EditResponse extends CoreEditResponse
 {
-    protected $files = array();
+    /**
+     * The added/modified file versions.
+     *
+     * @var \Concrete\Core\Entity\File\File[]|\Concrete\Core\Entity\File\Version[]
+     */
+    protected $files = [];
 
+    /**
+     * Add a file to the added/modified file list.
+     *
+     * @param \Concrete\Core\Entity\File\File $file
+     *
+     * @return $this
+     */
     public function setFile(FileEntity $file)
     {
-        $this->files[] = $file;
+        $this->files[] = $file->getRecentVersion();
```

Comment:
Couldn't this be a problem? What if the user is expecting a file when they get files from this response?

Reply:
Well, there's no public way to access the `$files` protected property other than via `getJSONObject`.
`getJSONObject` calls `getJSONObject` on every item of `$files`.
Since the `File` entity doesn't have a `getJSONObject` method, we [https://github.com/concrete5/concrete5/blob/27bcd1b36ef35f21b0df32bfd7e57d2ecf347d65/concrete/src/Entity/File/File.php#L102](call) the `getJSONObject` method of the approved file version.

That is (I think) the reason why I made this change. But I agree that it's a breaking change (someone may subclass this EditResponse class, or pass a subclass of `File` to the method, so I reverted this change.